src/Avalonia.Controls/Control.cs

```diff
@@ -329,5 +347,55 @@ protected override void OnKeyUp(KeyEventArgs e)
                 }
             }
         }
+
+        protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T> change)
+        {
+            base.OnPropertyChanged(change);
+
+            if (change.Property == FlowDirectionProperty)
```

Comment:
I think we don't need to handle children here because FlowDirection is inherited and if the parent's value changes the child's value changes as well

Reply:
No, If the parent has changed and the child is permanent, the value will not be inherited and the child will be invalid. Consider the following case:
```
<Panel FlowDirection="RightToLeft">
  <TextBlock FlowDirection="LeftToRight">
</Panel>
```
if i change the `Panel` and he don't notify to `TextBlock` (because he constant FlowDirection), he is get wrong.

Reply:
I guess we can keep the invalidation of children then

Reply:
There is some double work with this implementation (albeit negligible), if you want, I can implement it a little differently so that a property change will call the InvalidateFlowDirection method for itself, and its children for another method such as `ParentFlowDirectionChange` and then we can mark that `IsMirrorTransform` has already been handled.

Reply:
Keep it as is

Reply:
ok